src/coreclr/jit/assertionprop.cpp

```diff
@@ -2562,6 +2562,14 @@ GenTree* Compiler::optVNConstantPropOnTree(BasicBlock* block, GenTree* tree)
     ValueNumPair vnPair = tree->gtVNPair;
     ValueNum     vnCns  = vnStore->VNConservativeNormalValue(vnPair);
 
+    // Replace KEEPALIVE(ObjHandle) with NO-OP as the object handle is always
+    // kept alive by the runtime.
+    if (tree->OperIs(GT_KEEPALIVE))
+    {
+        const ValueNumPair opVnp = tree->gtGetOp1()->gtVNPair;
+        return opVnp.BothEqual() && vnStore->IsVNObjHandle(opVnp.GetLiberal()) ? gtNewNothingNode() : nullptr;
```

Comment:
I think you can just check the conservative VN.

Reply:
Addressed